File: tests/icare_hdf_reader/C/example/read_hdf_file_structure.c
/**
 * @file read_hdf_file_structure.c
 * @brief How to read the structure of an HDF4 file : retrieve the names of the SDS and the names of the file attributes
 * @author Nicolas PASCAL (nicolas.pascal@icare.univ-lille1.fr), (C) Centre de Gestion et de Traitement de Données (CGTD) ICARE 2008
 * @version 0.0.0
 * @date 2008/04/07
 *
 * Copyright: See COPYING file that comes with this distribution
 *
 * History :
 *   v0.0.0 : creation
 */

#include "c_hdf_reader.h"

int main( int argc, char**argv ) {
    /* generic iterator */
    int i = 0;
    /* path to the file to read */
    char filename[] = "../data/test.hdf";
    /* return status */
    intn status    = -1;
    /* number of SDS of the file */
    int32 n_sds = -1;
    /* list of the SDS names of a file. Set it to NULL to let the library does all memory allocations */
    char ** v_sds_name = NULL;
    /* number of file attibutes */
    int32 n_file_attr = -1;
    /* list of the attribute names of a file. Set it to NULL to let the library does all memory allocations */
    char ** v_file_attr_name = NULL;

    /**********************************************/
    /* RETRIEVE HDF STRUCTURE :                   */
    /* - LIST OF SDS NAMES and                    */
    /* - LIST OF FILE ATTRIBUTES NAMES            */
    /**********************************************/
    status = get_file_info(filename, &n_sds, &v_sds_name, &n_file_attr, &v_file_attr_name);
    if (status == FAIL) // retrieve infos error
        return FAIL;

    /* print them out */
    printf("*** Structure of file %s ***\n",filename);
    /* -> sds */
    printf("--- SDS ---\n");
    for ( i = 0 ; i < n_sds ; ++i )
        printf("\t%s\n", v_sds_name[i]);
    /* -> file attributes */
    printf("--- FILE ATTRIBUTES ---\n");
    for ( i = 0 ; i < n_file_attr ; ++i )
        printf("\t%s\n",v_file_attr_name[i]);
    printf("\n");

    /* free allocations made internally by get_file_info */
    free_v_string(&v_sds_name, n_sds);
    free_v_string(&v_file_attr_name, n_file_attr);

    return 0;
}

File: tests/1/icare_hdf_reader/PYTHON/example/read_hdf_file_structure.py
#! /usr/bin/env python
# -*- coding: latin-1 -*-

###################################################################################################################################
#
# @file read_hdf_file_structure.py
# @brief How to read the structure of an HDF4 file : retrieve the names of the SDS and the names of the file attributes
# @author Nicolas PASCAL (nicolas.pascal@icare.univ-lille1.fr), (C) Centre de Gestion et de Traitement de Données (CGTD) ICARE 2008
# @version 0.0.0
# @date 2008/04/07
#
# Copyright: See COPYING file that comes with this distribution
#
# History :
#   v0.0.0 : creation
#
###################################################################################################################################

from py_hdf_reader import *

__VERSION__="0.0.0"
__AUTHOR__="CGTD/UDEV Nicolas PASCAL (nicolas.pascal@icare.univ-lille1.fr)"

# path to the file to read
# filename = "../data/test.hdf";
filename = "MOD35_L2.A2017001.1355.061.2017312070506.hdf"

#*********************************************/
# RETRIEVE HDF STRUCTURE :
# - LIST OF SDS NAMES and
# - LIST OF FILE ATTRIBUTES NAMES
#*********************************************/

# --- read HDF file structure ---
n_sds, v_sds_name, n_file_attr, v_file_attr_name = get_file_info(filename)
# --- print it out ---
print("*** Structure of file %s ***"%filename)
# -> sds
print("--- SDS ---");
for i in range(n_sds) :
    print("\t%s"%v_sds_name[i])
# -> file attributes
print("--- FILE ATTRIBUTES ---")
for i in range(n_file_attr) :
    print("\t%s"%v_file_attr_name[i])
print("\n")